pkg/app/app.go

```diff
@@ -101,65 +116,80 @@ func (a *App) Run() error {
 	klog.Info("starting main application loop")
 	for {
 		if err := a.refreshVaultTokenFromFile(); err != nil {
-			klog.Error(err)
+			return err
+		}
+		var wg sync.WaitGroup
+		for _, workspace := range a.Config.TFCloud {
+
+			wg.Add(1)
+			go a.updateTFCloudInstance(workspace, &wg)
+		}
+		for _, project := range a.Config.CircleCI {
+			wg.Add(1)
+			go a.updateCircleCIInstance(project, &wg)
 		}
-		a.updateCircleCI()
-		a.updateTFCloud()
+		wg.Wait()
```

Comment:
Woo! I had this thought last night after we were looking at the logs: "It would be nice if we used goroutines to do all these updates"

Looks great.